### PR TITLE
Add support for scikit-learn 1.5.0

### LINKED ISSUE #415: Add support for scikit-learn 1.5.0
### Problem Description
- Currently, we restrict scikit-learn <1.5 for python 3.10 and 3.11.
  - scikit-learn supports < 1.3.2 for python 3.8 and 3.9
- We would like to support the newest release of scikit-learn. 

### Expected behavior
- Copulas supports scikit-learn 1.50

### Additional context
- Remove upper bound restriction in pyproject.toml
- Update any associated tests

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86b10kbeh
Resolve #415

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b10kbeh Copulas - Add support for scikit-learn 1.5.0 #415](https://app.clickup.com/t/86b10kbeh)

**Review by gsheni — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -50,8 +50,7 @@ version = {attr = 'copulas.__version__'}
 [project.optional-dependencies]
 tutorials = [
     'markupsafe<=2.0.1',
-    "scikit-learn>=0.24,<1.3.2;python_version<'3.10'",
-    "scikit-learn>=0.24,<1.5;python_version>='3.10' and python_version<'3.12'",
+    "scikit-learn>=0.24;python_version<'3.12'",
```
Comment (amontanez24) [inline pyproject.toml:53]: do we know why it was failing before? @gsheni

Comment (gsheni) [reply pyproject.toml:53]: I do not. We put this restriction when we added Python 3.12 support:
https://github.com/sdv-dev/Copulas/commit/1c088f3c0a8815c8a2a5bc0c25adc6b280952221#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L50

Comment (amontanez24) [reply pyproject.toml:53]: @fealho do you remember?

Comment (fealho) [reply pyproject.toml:53]: @amontanez24 I don't think anything was failing. When I put that cap [4 months ago](https://github.com/sdv-dev/Copulas/pull/386/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R54), the standard was to cap everything to one version above the latest available one, which was 1.5 [back in April](https://pypi.org/project/scikit-learn/#history).

Comment (amontanez24) [reply pyproject.toml:53]: the standard for libraries that have a 1.0 release is the next major version so it should've been 2.0

**Review by amontanez24 — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (fealho) [reply pyproject.toml:53]: When I made that change, scikit-learn was capped to 1.2, so I probably just updated it to the latest version, 1.5 at the time.

**Pull request merged**